### ISSUE TITLE
IOError when respond with large body

### ISSUE THREAD

**Issue by li-thy-um** [midori-rb/midori.rb#155]
This will cause the server process abort.

```
I, [2018-02-28T08:09:48.730775 #1]  INFO -- : Midori 0.5.2 is now running on 0.0.0.0:3000
I, [2018-02-28T08:09:50.043817 #1]  INFO -- : 172.25.0.1 - - "GET /documents HTTP/1.0" 200 OK 0.034884
I, [2018-02-28T08:09:50.360293 #1]  INFO -- : 172.25.0.1 - - "GET /origin_websites HTTP/1.0" 200 OK 0.286312
I, [2018-02-28T08:09:50.437701 #1]  INFO -- : 172.25.0.1 - - "GET /output_columns HTTP/1.0" 200 OK 0.076205
I, [2018-02-28T08:09:51.133797 #1]  INFO -- : 172.25.0.1 - - "GET /compose_organizations HTTP/1.0" 200 OK 0.693688
/app/vendor/bundle/ruby/2.5.0/gems/murasaki-0.2.0/lib/murasaki/event_loop.rb:62:in `fileno': closed stream (IOError)
	from /app/vendor/bundle/ruby/2.5.0/gems/murasaki-0.2.0/lib/murasaki/event_loop.rb:62:in `deregister'
	from /app/vendor/bundle/ruby/2.5.0/gems/midori.rb-0.5.2/lib/midori/connection.rb:63:in `close_connection'
	from /app/vendor/bundle/ruby/2.5.0/gems/midori.rb-0.5.2/lib/midori/connection.rb:33:in `block in listen'
	from /app/vendor/bundle/ruby/2.5.0/gems/murasaki-0.2.0/lib/murasaki/event_loop.rb:75:in `block in run_once'
	from /app/vendor/bundle/ruby/2.5.0/gems/murasaki-0.2.0/lib/murasaki/event_loop.rb:74:in `select'
	from /app/vendor/bundle/ruby/2.5.0/gems/murasaki-0.2.0/lib/murasaki/event_loop.rb:74:in `run_once'
	from /app/vendor/bundle/ruby/2.5.0/gems/murasaki-0.2.0/lib/murasaki/event_loop.rb:101:in `start'
	from /app/vendor/bundle/ruby/2.5.0/gems/midori.rb-0.5.2/lib/midori/runner.rb:44:in `start'
	from /app/midori.rb:6:in `<main>'
```

**Comment by dsh0416** [comment]
Bug confirmed. Can you provide any sample code?

**Comment by li-thy-um** [comment]
It's hard to provide sample code, but I'm trying.

I removed the Nginx reverse proxy in between, and did the test again.
It will response the data successfully, but the server will shutdown.

```ruby
get '/multi_aggs' do
   # here it will do elasticsearch aggregation search
   MultiAggs::Index.new(request.query_params, user).to_json
end

## I use rest-client do the request

resp = RestClient::Request.execute(method: :get, url: url,
      timeout: 5,
      headers: {params: params, Authorization: @auth_token})
```

```
I, [2018-03-01T04:48:23.105229 #1]  INFO -- : Midori 0.5.2 is now running on 0.0.0.0:3000
I, [2018-03-01T04:48:37.975727 #1]  INFO -- : 172.25.0.1 - - "GET /documents HTTP/1.0" 200 OK 0.257697
287465 # This is the response body's size
I, [2018-03-01T04:48:38.949328 #1]  INFO -- : 172.25.0.1 - - "GET /multi_aggs HTTP/1.1" 200 OK 0.971632
I, [2018-03-01T04:48:45.947080 #1]  INFO -- : 172.25.0.1 - - "GET /documents HTTP/1.0" 200 OK 0.330976
320426
I, [2018-03-01T04:48:47.213653 #1]  INFO -- : 172.25.0.1 - - "GET /multi_aggs HTTP/1.1" 200 OK 1.264781
I, [2018-03-01T04:48:55.243512 #1]  INFO -- : 172.25.0.1 - - "GET /documents HTTP/1.0" 200 OK 0.524588
361135
I, [2018-03-01T04:48:56.695555 #1]  INFO -- : 172.25.0.1 - - "GET /multi_aggs HTTP/1.1" 200 OK 1.449826
I, [2018-03-01T04:49:04.998189 #1]  INFO -- : 172.25.0.1 - - "GET /documents HTTP/1.0" 200 OK 0.205967
286526
I, [2018-03-01T04:49:06.100274 #1]  INFO -- : 172.25.0.1 - - "GET /multi_aggs HTTP/1.1" 200 OK 1.100490
I, [2018-03-01T04:49:10.953088 #1]  INFO -- : 172.25.0.1 - - "GET /documents HTTP/1.0" 200 OK 0.143179
180645
I, [2018-03-01T04:49:11.421074 #1]  INFO -- : 172.25.0.1 - - "GET /multi_aggs HTTP/1.1" 200 OK 0.465922
/app/vendor/bundle/ruby/2.5.0/gems/murasaki-0.2.0/lib/murasaki/event_loop.rb:62:in `fileno': closed stream (IOError)
	from /app/vendor/bundle/ruby/2.5.0/gems/murasaki-0.2.0/lib/murasaki/event_loop.rb:62:in `deregister'
	from /app/vendor/bundle/ruby/2.5.0/gems/midori.rb-0.5.2/lib/midori/connection.rb:63:in `close_connection'
	from /app/vendor/bundle/ruby/2.5.0/gems/midori.rb-0.5.2/lib/midori/connection.rb:33:in `block in listen'
	from /app/vendor/bundle/ruby/2.5.0/gems/murasaki-0.2.0/lib/murasaki/event_loop.rb:75:in `block in run_once'
	from /app/vendor/bundle/ruby/2.5.0/gems/murasaki-0.2.0/lib/murasaki/event_loop.rb:74:in `select'
	from /app/vendor/bundle/ruby/2.5.0/gems/murasaki-0.2.0/lib/murasaki/event_loop.rb:74:in `run_once'
	from /app/vendor/bundle/ruby/2.5.0/gems/murasaki-0.2.0/lib/murasaki/event_loop.rb:101:in `start'
	from /app/vendor/bundle/ruby/2.5.0/gems/midori.rb-0.5.2/lib/midori/runner.rb:44:in `start'
	from /app/midori.rb:6:in `<main>'
```

**Comment by li-thy-um** [comment]
Gems:

```ruby
gem 'activesupport'
gem 'tzinfo-data'
gem 'midori.rb'
gem 'elasticsearch'
gem 'monadic'
gem 'jbuilder'
gem 'nokogiri'
gem 'redis'
```

**Comment by dsh0416** [comment]
This seems to be the problem that midori accidentally closed the stream multiple times. A patch would be applied soon.

**Comment by luikore** [comment]
Any update on this bug?

**Comment by dsh0416** [comment]
@luikore There has been a temp fix merged, which has already fixed this problem. But it may take some time to rewrite the part with the fd number to fix the problem more elegantly.

**Comment by dsh0416** [comment]
Fixed in the v0.5.4 with the update of [murasaki](https://github.com/midori-rb/murasaki).

Thank you guys so much for reporting and tracking this issue. Sorry for the late fix.
